### PR TITLE
Fix FP S2325: Event Handlers should not be static

### LINKED ISSUE #5536: S2325 should not be raised on Event Handlers.
### Description

Currently S2325 is raised on form event handlers that do not access any form state: 
> Methods and properties that don't access instance data should be static

### Repro steps

```
private void arbitraryControl_DragEnter(object sender, DragEventArgs e)
{
    e.Effect = DragDropEffects.Link;
}
```

### Expected behavior

No issue raised as Windows Forms designer cannot handle static event handlers.

### Actual behavior

S2325 is raised.

### Known workarounds

Suppression attribute added to all affected methods.

### PR DESCRIPTION
**Pull request by Corniel**
Given the required scenario, I opted for a separate test case.

Fixes #5536

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs [hunk 1]
```diff
@@ -20,11 +20,13 @@
 
 using System;
 using System.Collections.Immutable;
+using System.Diagnostics;
 using System.Linq;
 using Microsoft.CodeAnalysis;
 
 namespace SonarAnalyzer.Helpers
 {
+    [DebuggerDisplay("{TypeName}")]
```
Comment (costin-zaharia-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs:29]: Not needed after rebase.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MemberShouldBeStatic.WinForms.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        public Control ActiveControl { get; set; }
+
+        public bool ActivateControl(Control active) { return true; }
+
+        private void MethodWith1Argument(object sender) { Handle(); } // Noncompliant
+
+        private void MethodWith3Argument(object sender, MyEventArgs e, string other) { Handle(); } // Noncompliant
+        
+        private void SenderArgumentNotObject(string input, MyEventArgs e) { Handle(); }// Noncompliant
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MemberShouldBeStatic.WinForms.cs:16]: pedantic: insert a space between comment and method body

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]
So far, it looks good to me. I've added a few minor comments and it needs a rebase.

**Comment by Corniel** [comment]
Rebase done. And the two nitpicking issues.

**Comment by Corniel** [comment]
@costin-zaharia-sonarsource What do I have to change? As far as I'm concerned, the rebase have been applied.

**Comment by costin-zaharia-sonarsource** [comment]
> @costin-zaharia-sonarsource What do I have to change? As far as I'm concerned, the rebase have been applied.

Nothing. I was waiting for the build to finish.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**